src/features/todosSlice.js

```diff
@@ -0,0 +1,58 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import { createSlice } from '@reduxjs/toolkit';
+
+const KEY_TODO = 'KEY_TODO';
+
+const todoDoneReducer = (state, value) => {
+  const { id, done } = value.payload;
+  const newTodos = state.map(obj => {
+    if (obj.id === id) {
+      return { ...obj, done: done };
+    }
+    return obj;
+  });
+  localStorage.setItem(KEY_TODO, JSON.stringify(newTodos));
```

Comment:
Reducers should not have side effects. Move interactions with localStorage to Redux middleware , please